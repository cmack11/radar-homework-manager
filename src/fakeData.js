import moment from 'moment'

export const subjects2 = [];
export const subjects1 = [
  			{
  				name:"Subject #1",
  				color:"blue",
  				assignments:[
  					{
  						name:"S1A1",
  						type:"Assignment",
						dueDate:moment().add(3,'days'),
						subject:"Subject #1",
  					},{
  						name:"S1PS2",
  						type:"Problem Set",
  						dueDate:moment().add(9,'days'),
						subject:"Subject #1",
  					},{
  						name:"S1A3",
  						type:"Assignment",
  						dueDate:moment().add(5,'days'),
						subject:"Subject #1",
  					},{
  						name:"S1PS4",
  						type:"Problem Set",
  						dueDate:moment().add(8,'days'),
						subject:"Subject #1",
  					},{
  						name:"S1A5",
  						type:"Assignment",
  						dueDate:moment().add(1,'seconds'),
						subject:"Subject #1",
  					},{
  						name:"S1R6",
  						type:"Reading",
  						dueDate:moment().add(10,'days').add(5,'hours'),
						subject:"Subject #1",
  					},{
  						name:"S1A7",
  						type:"Assignment",
  						dueDate:moment().add(13,'days').subtract(3,'hours'),
						subject:"Subject #1",
  					},{
  						name:"S1A8",
  						type:"Assignment",
  						dueDate:moment().add(4,'days').subtract(1,'hours'),
						subject:"Subject #1",
  					}
  				]
  			},{
  				name:"Subject #2",
  				color:"violet",
  				assignments:[
  					{
  						name:"S2A1",
  						type:"Assignment",
  						dueDate:moment().add(7,'days'),
						subject:"Subject #2",
  					},{
  						name:"S2E2",
  						type:"Exam",
  						dueDate:moment().add(4,'days'),
						subject:"Subject #2",
  					},{
  						name:"S2A3",
  						type:"Assignment",
  						dueDate:moment().add(13,'days').add(12,'hours'),
						subject:"Subject #2",
  					},{
  						name:"S2A4",
  						type:"Assignment",
  						dueDate:moment().add(3,'days').add(20,'hours'),
						subject:"Subject #2",
  					},{
  						name:"S2A5",
  						type:"Assignment",
  						dueDate:moment().add(1,'seconds'),
						subject:"Subject #2",
  					}
  				]
  			},{
  				name:"Subject #3",
  				color:"orange",
  				assignments:[
  					{
  						name:"S3A1",
  						type:"Assignment",
  						dueDate:moment().add(5,'days'),
						subject:"Subject #3",
  					},{
  						name:"S3PS2",
  						type:"Problem Set",
  						dueDate:moment().add(4,'days'),
						subject:"Subject #3",
  					},{
  						name:"S3E3",
  						type:"Exam",
  						dueDate:moment().add(8,'days').add(12,'hours'),
						subject:"Subject #3",
  					},{
  						name:"S3A4",
  						type:"Assignment",
  						dueDate:moment().add(4,'days'),
						subject:"Subject #3",
  					},{
  						name:"S3A5",
  						type:"Assignment",
  						dueDate:moment().add(13,'days').add(12,'hours'),
						subject:"Subject #3",
  					},{
  						name:"S3A6",
  						type:"Assignment",
  						dueDate:moment().add(1,'seconds'),
						subject:"Subject #3",
  					}

  				]
  			}, {
  				name:"Subject #4",
  				color:"purple",
  				assignments:[
  					{
  						name:"S4A1",
  						type:"Assignment",
						dueDate:moment().add(6,'days'),
						subject:"Subject #4",
  					},{
  						name:"S4R2",
  						type:"Reading",
  						dueDate:moment().add(9,'days'),
						subject:"Subject #4",
  					},{
  						name:"S4A3",
  						type:"Assignment",
  						dueDate:moment().add(2,'days').subtract(1,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A4",
  						type:"Assignment",
  						dueDate:moment().add(1,'days').add(4,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A5",
  						type:"Assignment",
  						dueDate:moment().add(12,'days').add(9,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A6",
  						type:"Assignment",
  						dueDate:moment().add(5,'days').add(4,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A7",
  						type:"Assignment",
  						dueDate:moment().add(4,'days').add(9,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A8",
  						type:"Assignment",
  						dueDate:moment().add(11,'days').add(4,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4E9",
  						type:"Exam",
  						dueDate:moment().add(3,'days').add(9,'hours'),
						subject:"Subject #4",
  					},{
  						name:"S4A10",
  						type:"Assignment",
  						dueDate:moment().add(1,'seconds'),
						subject:"Subject #4",
  					}
  				]
  			},{
  				name:"Subject #5",
  				color:"darkgreen",
  				assignments:[
  					{
  						name:"S5R1",
  						type:"Reading",
  						dueDate:moment().add(2,'days'),
						subject:"Subject #5",
  					},{
  						name:"S5A2",
  						type:"Assignment",
  						dueDate:moment().add(8,'days'),
						subject:"Subject #5",
  					},{
  						name:"S5A3",
  						type:"Assignment",
  						dueDate:moment().add(2,'days').subtract(1,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5A4",
  						type:"Assignment",
  						dueDate:moment().add(11,'days').add(4,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5R5",
  						type:"Reading",
  						dueDate:moment().add(5,'days').add(3,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5A6",
  						type:"Assignment",
  						dueDate:moment().add(10,'days').add(4,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5R7",
  						type:"Reading",
  						dueDate:moment().add(6,'days').add(9,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5A8",
  						type:"Assignment",
  						dueDate:moment().add(12,'days').add(4,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5E9",
  						type:"Exam",
  						dueDate:moment().add(6,'days').add(9,'hours'),
						subject:"Subject #5",
  					},{
  						name:"S5A10",
  						type:"Assignment",
  						dueDate:moment().add(7,'days').subtract(30,'minutes'),
						subject:"Subject #5",
  					},{
  						name:"S5A11",
  						type:"Assignment",
  						dueDate:moment().add(1,'seconds'),
						subject:"Subject #5",
  					}
  				]
  			}
  	];

export const colors1 = {
	typeColors:{
		"Assignment":'white',
		"Exam":'maroon',
		"Reading":'cyan',
		"Problem Set":'magenta'
	}
}
