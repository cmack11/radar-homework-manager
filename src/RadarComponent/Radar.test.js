import React from 'react';
import {Radar} from './Radar.js';
import { shallow, mount, render } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {subjects1} from '../fakeData.js';
import moment from 'moment'

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() })
  jest.useFakeTimers();
});

const subjects = [
  {
    name:'Subject #1',
    color:'red',
    assignments:[
      {
        name:"Assignment #1",
        type:"Problem Set",
        dueDate:moment('2018-09-12 12:00:00.000'),
      subject:"Subject #1",
      },{
        name:"Assignment #2",
        type:"Assignment",
        dueDate:moment('2018-09-13 14:00:00.000'),
        subject:"Subject #1",
      }
    ]
  },
  {
    name:'Subject #2',
    color:'blue',
    assignments:[]
  }
]

const getDistanceFromCenter = function() {return 1};
const view = {
  dotsView:{width:100,height:100},
  x:10,
  y:10,
  width:50,
  height:50,
  dotRadiusPercent:10,
  strokeWidth:2,
  ringsWidth:1000000,
  colors:{
    subjectDivider:'blue',
    typeColors:{Assignment:'green'}
  }
}
const dates = {
  startDate:moment('2018-09-10 14:00:00.000'),
  endDate:moment('2018-09-17 14:00:00.000')
};



describe('the initialization of the component', () => {

  it('creates default Radar component', () => {
    const wrapper = shallow(
      <Radar  subjects={[]} 
          view={{dotsView:{width:100,height:100}}} 
          />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('creates a Radar component', () => {
    const wrapper = shallow(
      <Radar  subjects={subjects} 
          view={view} 
          dates={dates}
          />
    );

    //expect(wrapper).toMatchSnapshot();
  });
})


/*describe(' ', () => {

  it('should ', () => {
    const wrapper = shallow(
      <Radar  subjects={subjects} 
          getDistanceFromCenter={getDistanceFromCenter} 
          view={view} 
          dims={dims}
          />);
      let radarObj = wrapper.instance();


  });

})*/



