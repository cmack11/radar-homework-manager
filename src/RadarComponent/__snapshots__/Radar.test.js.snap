// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the initialization of the component creates default Radar component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Radar
    subjects={Array []}
    view={
      Object {
        "colors": Object {
          "subjectDivider": "#39ff14",
          "typeColors": Object {},
        },
        "dotRadiusPercent": 0.07,
        "dotsView": Object {
          "height": 100,
          "width": 100,
        },
        "height": 250,
        "ringsWidth": 604800000,
        "strokeWidth": 3,
        "width": 250,
        "x": 0,
        "y": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        height={100}
        id="radar"
        stroke="black"
        strokeWidth="2"
        width={100}
      >
        <svg
          height={250}
          stroke="#39ff14"
          strokeWidth={3}
          width={250}
          x={0}
          y={0}
        >
          <circle
            cx={125}
            cy={125}
            fill="none"
            r={122}
          />
          <circle
            cx={125}
            cy={125}
            fill="#39ff14"
            r={8.540000000000001}
          />
        </svg>
        <Dots
          dims={
            Object {
              "buttons": Object {
                "width": 50,
              },
              "dotViewer": Object {
                "height": 250,
                "width": 250,
              },
              "dots": Object {
                "center": Object {
                  "x": 125,
                  "y": 125,
                },
                "radius": 8.540000000000001,
              },
              "radar": Object {
                "center": Object {
                  "x": 125,
                  "y": 125,
                },
                "height": 250,
                "radius": 122,
                "width": 250,
                "x": 0,
                "y": 0,
              },
              "style": Object {
                "strokeColor": "#39ff14",
                "strokeWidth": 3,
              },
            }
          }
          getDistanceFromCenter={[Function]}
          intersectFunctions={
            Array [
              Object {
                "func": [Function],
                "rect": Object {
                  "height": 125,
                  "width": 125,
                  "x": 10,
                  "y": -35,
                },
              },
            ]
          }
          intersectsLine={[Function]}
          setClickedDot={[Function]}
          setDraggedDot={[Function]}
          subjects={Array []}
          view={
            Object {
              "colors": Object {
                "subjectDivider": "#39ff14",
                "typeColors": Object {},
              },
              "dotRadiusPercent": 0.07,
              "dotsView": Object {
                "height": 100,
                "width": 100,
              },
              "height": 250,
              "ringsWidth": 604800000,
              "strokeWidth": 3,
              "width": 250,
              "x": 0,
              "y": 0,
            }
          }
        />
        <svg
          height={250}
          stroke="#39ff14"
          strokeWidth={3}
          width={250}
          x={0}
          y={0}
        >
          <SpinLine
            center={
              Object {
                "x": 125,
                "y": 125,
              }
            }
            lineColor="#39ff14"
            radius={122}
            rpm={6}
            setLineAngle={[Function]}
            show={true}
          />
        </svg>
        <image
          height={50}
          href="add_button.png"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          width={50}
          x={40}
          y={40}
        />
        <image
          height={50}
          href="history_button.png"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
          width={50}
          x={10}
          y={40}
        />
        <DotViewer
          closeDotViewer={[Function]}
          height={200}
          width={250}
        />
        <DraggedDot
          intersectFunctions={
            Array [
              Object {
                "func": [Function],
                "rect": Object {
                  "height": 125,
                  "width": 125,
                  "x": 10,
                  "y": -35,
                },
              },
            ]
          }
          radius={12.810000000000002}
        />
      </svg>,
      "id": "radardiv",
      "style": Object {
        "position": "absolute",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <svg
            height={250}
            stroke="#39ff14"
            strokeWidth={3}
            width={250}
            x={0}
            y={0}
          >
            <circle
              cx={125}
              cy={125}
              fill="none"
              r={122}
            />
            <circle
              cx={125}
              cy={125}
              fill="#39ff14"
              r={8.540000000000001}
            />
          </svg>,
          Array [],
          <Dots
            dims={
              Object {
                "buttons": Object {
                  "width": 50,
                },
                "dotViewer": Object {
                  "height": 250,
                  "width": 250,
                },
                "dots": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "radius": 8.540000000000001,
                },
                "radar": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "height": 250,
                  "radius": 122,
                  "width": 250,
                  "x": 0,
                  "y": 0,
                },
                "style": Object {
                  "strokeColor": "#39ff14",
                  "strokeWidth": 3,
                },
              }
            }
            getDistanceFromCenter={[Function]}
            intersectFunctions={
              Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ]
            }
            intersectsLine={[Function]}
            setClickedDot={[Function]}
            setDraggedDot={[Function]}
            subjects={Array []}
            view={
              Object {
                "colors": Object {
                  "subjectDivider": "#39ff14",
                  "typeColors": Object {},
                },
                "dotRadiusPercent": 0.07,
                "dotsView": Object {
                  "height": 100,
                  "width": 100,
                },
                "height": 250,
                "ringsWidth": 604800000,
                "strokeWidth": 3,
                "width": 250,
                "x": 0,
                "y": 0,
              }
            }
          />,
          <svg
            height={250}
            stroke="#39ff14"
            strokeWidth={3}
            width={250}
            x={0}
            y={0}
          >
            <SpinLine
              center={
                Object {
                  "x": 125,
                  "y": 125,
                }
              }
              lineColor="#39ff14"
              radius={122}
              rpm={6}
              setLineAngle={[Function]}
              show={true}
            />
          </svg>,
          <image
            height={50}
            href="add_button.png"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            width={50}
            x={40}
            y={40}
          />,
          <image
            height={50}
            href="history_button.png"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            width={50}
            x={10}
            y={40}
          />,
          <DotViewer
            closeDotViewer={[Function]}
            height={200}
            width={250}
          />,
          <DraggedDot
            intersectFunctions={
              Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ]
            }
            radius={12.810000000000002}
          />,
        ],
        "height": 100,
        "id": "radar",
        "stroke": "black",
        "strokeWidth": "2",
        "width": 100,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <circle
                cx={125}
                cy={125}
                fill="none"
                r={122}
              />,
              <circle
                cx={125}
                cy={125}
                fill="#39ff14"
                r={8.540000000000001}
              />,
              Array [],
            ],
            "height": 250,
            "stroke": "#39ff14",
            "strokeWidth": 3,
            "width": 250,
            "x": 0,
            "y": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": 125,
                "cy": 125,
                "fill": "none",
                "r": 122,
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "cx": 125,
                "cy": 125,
                "fill": "#39ff14",
                "r": 8.540000000000001,
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
          ],
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dims": Object {
              "buttons": Object {
                "width": 50,
              },
              "dotViewer": Object {
                "height": 250,
                "width": 250,
              },
              "dots": Object {
                "center": Object {
                  "x": 125,
                  "y": 125,
                },
                "radius": 8.540000000000001,
              },
              "radar": Object {
                "center": Object {
                  "x": 125,
                  "y": 125,
                },
                "height": 250,
                "radius": 122,
                "width": 250,
                "x": 0,
                "y": 0,
              },
              "style": Object {
                "strokeColor": "#39ff14",
                "strokeWidth": 3,
              },
            },
            "getDistanceFromCenter": [Function],
            "intersectFunctions": Array [
              Object {
                "func": [Function],
                "rect": Object {
                  "height": 125,
                  "width": 125,
                  "x": 10,
                  "y": -35,
                },
              },
            ],
            "intersectsLine": [Function],
            "setClickedDot": [Function],
            "setDraggedDot": [Function],
            "subjects": Array [],
            "view": Object {
              "colors": Object {
                "subjectDivider": "#39ff14",
                "typeColors": Object {},
              },
              "dotRadiusPercent": 0.07,
              "dotsView": Object {
                "height": 100,
                "width": 100,
              },
              "height": 250,
              "ringsWidth": 604800000,
              "strokeWidth": 3,
              "width": 250,
              "x": 0,
              "y": 0,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SpinLine
              center={
                Object {
                  "x": 125,
                  "y": 125,
                }
              }
              lineColor="#39ff14"
              radius={122}
              rpm={6}
              setLineAngle={[Function]}
              show={true}
            />,
            "height": 250,
            "stroke": "#39ff14",
            "strokeWidth": 3,
            "width": 250,
            "x": 0,
            "y": 0,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "center": Object {
                "x": 125,
                "y": 125,
              },
              "lineColor": "#39ff14",
              "radius": 122,
              "rpm": 6,
              "setLineAngle": [Function],
              "show": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": 50,
            "href": "add_button.png",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
            "width": 50,
            "x": 40,
            "y": 40,
          },
          "ref": null,
          "rendered": null,
          "type": "image",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "height": 50,
            "href": "history_button.png",
            "onClick": [Function],
            "style": Object {
              "cursor": "pointer",
            },
            "width": 50,
            "x": 10,
            "y": 40,
          },
          "ref": null,
          "rendered": null,
          "type": "image",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "closeDotViewer": [Function],
            "dot": undefined,
            "height": 200,
            "width": 250,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "dot": undefined,
            "intersectFunctions": Array [
              Object {
                "func": [Function],
                "rect": Object {
                  "height": 125,
                  "width": 125,
                  "x": 10,
                  "y": -35,
                },
              },
            ],
            "radius": 12.810000000000002,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "svg",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          height={100}
          id="radar"
          stroke="black"
          strokeWidth="2"
          width={100}
        >
          <svg
            height={250}
            stroke="#39ff14"
            strokeWidth={3}
            width={250}
            x={0}
            y={0}
          >
            <circle
              cx={125}
              cy={125}
              fill="none"
              r={122}
            />
            <circle
              cx={125}
              cy={125}
              fill="#39ff14"
              r={8.540000000000001}
            />
          </svg>
          <Dots
            dims={
              Object {
                "buttons": Object {
                  "width": 50,
                },
                "dotViewer": Object {
                  "height": 250,
                  "width": 250,
                },
                "dots": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "radius": 8.540000000000001,
                },
                "radar": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "height": 250,
                  "radius": 122,
                  "width": 250,
                  "x": 0,
                  "y": 0,
                },
                "style": Object {
                  "strokeColor": "#39ff14",
                  "strokeWidth": 3,
                },
              }
            }
            getDistanceFromCenter={[Function]}
            intersectFunctions={
              Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ]
            }
            intersectsLine={[Function]}
            setClickedDot={[Function]}
            setDraggedDot={[Function]}
            subjects={Array []}
            view={
              Object {
                "colors": Object {
                  "subjectDivider": "#39ff14",
                  "typeColors": Object {},
                },
                "dotRadiusPercent": 0.07,
                "dotsView": Object {
                  "height": 100,
                  "width": 100,
                },
                "height": 250,
                "ringsWidth": 604800000,
                "strokeWidth": 3,
                "width": 250,
                "x": 0,
                "y": 0,
              }
            }
          />
          <svg
            height={250}
            stroke="#39ff14"
            strokeWidth={3}
            width={250}
            x={0}
            y={0}
          >
            <SpinLine
              center={
                Object {
                  "x": 125,
                  "y": 125,
                }
              }
              lineColor="#39ff14"
              radius={122}
              rpm={6}
              setLineAngle={[Function]}
              show={true}
            />
          </svg>
          <image
            height={50}
            href="add_button.png"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            width={50}
            x={40}
            y={40}
          />
          <image
            height={50}
            href="history_button.png"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
            width={50}
            x={10}
            y={40}
          />
          <DotViewer
            closeDotViewer={[Function]}
            height={200}
            width={250}
          />
          <DraggedDot
            intersectFunctions={
              Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ]
            }
            radius={12.810000000000002}
          />
        </svg>,
        "id": "radardiv",
        "style": Object {
          "position": "absolute",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <svg
              height={250}
              stroke="#39ff14"
              strokeWidth={3}
              width={250}
              x={0}
              y={0}
            >
              <circle
                cx={125}
                cy={125}
                fill="none"
                r={122}
              />
              <circle
                cx={125}
                cy={125}
                fill="#39ff14"
                r={8.540000000000001}
              />
            </svg>,
            Array [],
            <Dots
              dims={
                Object {
                  "buttons": Object {
                    "width": 50,
                  },
                  "dotViewer": Object {
                    "height": 250,
                    "width": 250,
                  },
                  "dots": Object {
                    "center": Object {
                      "x": 125,
                      "y": 125,
                    },
                    "radius": 8.540000000000001,
                  },
                  "radar": Object {
                    "center": Object {
                      "x": 125,
                      "y": 125,
                    },
                    "height": 250,
                    "radius": 122,
                    "width": 250,
                    "x": 0,
                    "y": 0,
                  },
                  "style": Object {
                    "strokeColor": "#39ff14",
                    "strokeWidth": 3,
                  },
                }
              }
              getDistanceFromCenter={[Function]}
              intersectFunctions={
                Array [
                  Object {
                    "func": [Function],
                    "rect": Object {
                      "height": 125,
                      "width": 125,
                      "x": 10,
                      "y": -35,
                    },
                  },
                ]
              }
              intersectsLine={[Function]}
              setClickedDot={[Function]}
              setDraggedDot={[Function]}
              subjects={Array []}
              view={
                Object {
                  "colors": Object {
                    "subjectDivider": "#39ff14",
                    "typeColors": Object {},
                  },
                  "dotRadiusPercent": 0.07,
                  "dotsView": Object {
                    "height": 100,
                    "width": 100,
                  },
                  "height": 250,
                  "ringsWidth": 604800000,
                  "strokeWidth": 3,
                  "width": 250,
                  "x": 0,
                  "y": 0,
                }
              }
            />,
            <svg
              height={250}
              stroke="#39ff14"
              strokeWidth={3}
              width={250}
              x={0}
              y={0}
            >
              <SpinLine
                center={
                  Object {
                    "x": 125,
                    "y": 125,
                  }
                }
                lineColor="#39ff14"
                radius={122}
                rpm={6}
                setLineAngle={[Function]}
                show={true}
              />
            </svg>,
            <image
              height={50}
              href="add_button.png"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
              width={50}
              x={40}
              y={40}
            />,
            <image
              height={50}
              href="history_button.png"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
              width={50}
              x={10}
              y={40}
            />,
            <DotViewer
              closeDotViewer={[Function]}
              height={200}
              width={250}
            />,
            <DraggedDot
              intersectFunctions={
                Array [
                  Object {
                    "func": [Function],
                    "rect": Object {
                      "height": 125,
                      "width": 125,
                      "x": 10,
                      "y": -35,
                    },
                  },
                ]
              }
              radius={12.810000000000002}
            />,
          ],
          "height": 100,
          "id": "radar",
          "stroke": "black",
          "strokeWidth": "2",
          "width": 100,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <circle
                  cx={125}
                  cy={125}
                  fill="none"
                  r={122}
                />,
                <circle
                  cx={125}
                  cy={125}
                  fill="#39ff14"
                  r={8.540000000000001}
                />,
                Array [],
              ],
              "height": 250,
              "stroke": "#39ff14",
              "strokeWidth": 3,
              "width": 250,
              "x": 0,
              "y": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "cx": 125,
                  "cy": 125,
                  "fill": "none",
                  "r": 122,
                },
                "ref": null,
                "rendered": null,
                "type": "circle",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "cx": 125,
                  "cy": 125,
                  "fill": "#39ff14",
                  "r": 8.540000000000001,
                },
                "ref": null,
                "rendered": null,
                "type": "circle",
              },
            ],
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dims": Object {
                "buttons": Object {
                  "width": 50,
                },
                "dotViewer": Object {
                  "height": 250,
                  "width": 250,
                },
                "dots": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "radius": 8.540000000000001,
                },
                "radar": Object {
                  "center": Object {
                    "x": 125,
                    "y": 125,
                  },
                  "height": 250,
                  "radius": 122,
                  "width": 250,
                  "x": 0,
                  "y": 0,
                },
                "style": Object {
                  "strokeColor": "#39ff14",
                  "strokeWidth": 3,
                },
              },
              "getDistanceFromCenter": [Function],
              "intersectFunctions": Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ],
              "intersectsLine": [Function],
              "setClickedDot": [Function],
              "setDraggedDot": [Function],
              "subjects": Array [],
              "view": Object {
                "colors": Object {
                  "subjectDivider": "#39ff14",
                  "typeColors": Object {},
                },
                "dotRadiusPercent": 0.07,
                "dotsView": Object {
                  "height": 100,
                  "width": 100,
                },
                "height": 250,
                "ringsWidth": 604800000,
                "strokeWidth": 3,
                "width": 250,
                "x": 0,
                "y": 0,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SpinLine
                center={
                  Object {
                    "x": 125,
                    "y": 125,
                  }
                }
                lineColor="#39ff14"
                radius={122}
                rpm={6}
                setLineAngle={[Function]}
                show={true}
              />,
              "height": 250,
              "stroke": "#39ff14",
              "strokeWidth": 3,
              "width": 250,
              "x": 0,
              "y": 0,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "center": Object {
                  "x": 125,
                  "y": 125,
                },
                "lineColor": "#39ff14",
                "radius": 122,
                "rpm": 6,
                "setLineAngle": [Function],
                "show": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "svg",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": 50,
              "href": "add_button.png",
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
              "width": 50,
              "x": 40,
              "y": 40,
            },
            "ref": null,
            "rendered": null,
            "type": "image",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "height": 50,
              "href": "history_button.png",
              "onClick": [Function],
              "style": Object {
                "cursor": "pointer",
              },
              "width": 50,
              "x": 10,
              "y": 40,
            },
            "ref": null,
            "rendered": null,
            "type": "image",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "closeDotViewer": [Function],
              "dot": undefined,
              "height": 200,
              "width": 250,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "dot": undefined,
              "intersectFunctions": Array [
                Object {
                  "func": [Function],
                  "rect": Object {
                    "height": 125,
                    "width": 125,
                    "x": 10,
                    "y": -35,
                  },
                },
              ],
              "radius": 12.810000000000002,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "svg",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
